src/plugins/d3/renderer/hooks/useShapes/pie.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import {arc, pie, select} from 'd3';
+import type {PieArcDatum} from 'd3';
+
+import type {PieSeries, PieSeriesData} from '../../../../../types/widget-data';
+import {block} from '../../../../../utils/cn';
+
+import {isStringValueInPercent, isStringValueInPixel} from '../../utils';
+import type {OnSeriesMouseLeave, OnSeriesMouseMove} from '../useTooltip/types';
+
+type PreparePieSeriesArgs = {
+    boundsWidth: number;
+    boundsHeight: number;
+    series: PieSeries;
+    svgContainer: SVGSVGElement | null;
+    onSeriesMouseMove?: OnSeriesMouseMove;
+    onSeriesMouseLeave?: OnSeriesMouseLeave;
+};
+
+const b = block('d3-pie');
+const DEFAULT_INNER_RADIUS = 0;
+
+const getRadius = (radiusRelatedToChart: number, style?: string | number) => {
+    if (typeof style === 'undefined') {
+        return radiusRelatedToChart;
+    }
+
+    if (typeof style === 'string') {
+        if (isStringValueInPercent(style)) {
+            const percentage = Number.parseFloat(style) / 100;
+            return radiusRelatedToChart * percentage;
+        }
+
+        if (isStringValueInPixel(style)) {
+            return Number.parseFloat(style);
+        }
+
+        // In case of incorrect style value
+        return radiusRelatedToChart;
+    }
+
+    return style;
+};
+
+const getInnerRadius = (radius: number, style?: string | number) => {
```

Comment:
Looks almost the same as the code above.
Maybe it is possible to pass the default value for innerRadius and merge the functions?